relentless/optimize/method.py

```diff
@@ -314,35 +323,78 @@ def __init__(self, abs_tol, max_iter, step_size, line_search=None):
         self.step_size = step_size
         self.line_search = line_search
 
+    def descent_amount(self, gradient):
+        r"""Calculate the descent amount for the optimization.
+
+        The descent amount is:
+
+        .. math::
+
+            d = \alpha\lVert\nabla f\left(\mathbf{x}\right)\rVert
```

Comment:
Do we want to use a different variable here since we have been using d for the step direction?